src/javascripts/components/cookies-page.mjs

```diff
@@ -1,28 +1,47 @@
 import { getConsentCookie, setConsentCookie } from './cookie-functions.mjs'
 
 class CookiesPage {
+  /**
+   * @param {Element} $module - HTML element
+   */
   constructor ($module) {
-    this.$module = $module
+    if (!($module instanceof HTMLElement)) {
+      return this
+    }
+
+    const $cookieForm = $module.querySelector('.js-cookies-page-form')
+    if (!($cookieForm instanceof HTMLFormElement)) {
+      return this
+    }
+
+    /** @satisfies {NodeListOf<HTMLFieldSetElement>} */
+    const $cookieFormFieldsets = $cookieForm.querySelectorAll('.js-cookies-page-form-fieldset')
+    const $cookieFormButton = $cookieForm.querySelector('.js-cookies-form-button')
+
+    const $successNotification = $module.querySelector('.js-cookies-page-success')
+    if (!($successNotification instanceof HTMLElement)) {
+      return this
```

Comment:
I guess we might follow up and change this to throw errors, but at the minute this has potential to trip us up by silently swallowing errors.

Given that `$successNotification` is only used within `showSuccessNotification`, could we move the lookup there, and return early if the banner isn't present?

That way at least the rest of the cookie page functionality will continue to work if the banner is missing for some reason.

Reply:
Good thinking

I've copied what we do elsewhere for _optional_ selectors

```mjs
const $successNotification = $module.querySelector('.js-cookies-page-success')
if ($successNotification instanceof HTMLElement) {
  this.$successNotification = $successNotification
}
```

☝️ Avoid appending to class property unless checks pass, but check again in `showSuccessNotification()`